api/src/main/java/io/strimzi/api/kafka/model/kafka/cruisecontrol/ApiUsers.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.kafka.cruisecontrol;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import io.strimzi.api.kafka.model.common.Constants;
+import io.strimzi.api.kafka.model.common.PasswordSource;
+import io.strimzi.api.kafka.model.common.UnknownPropertyPreserving;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static io.strimzi.api.kafka.model.kafka.cruisecontrol.HashLoginServiceApiUsers.TYPE_HASH_LOGIN_SERVICE;
+
+/**
+ * Abstract baseclass for different representations of Cruise Control's API users config, discriminated by {@link #getType() type}.
+ */
+@Buildable(
+        editableEnabled = false,
+        builderPackage = Constants.FABRIC8_KUBERNETES_API
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({"type", "valueFrom"})
+@EqualsAndHashCode
+public abstract class ApiUsers implements UnknownPropertyPreserving {
+    private String type;
+    private PasswordSource valueFrom;
+    private Map<String, Object> additionalProperties = new HashMap<>(0);
+
+    @Description("Type of the Cruise Control API users configuration. "
+            + "Supported format is: " + "`" + TYPE_HASH_LOGIN_SERVICE + "`")
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    @JsonProperty(required = true)
+    public String getType() {
+        return type;
+    }
+
+    public void setType(String type) {
+        this.type = type;
+    }
+
+    @Description("Secret from which the custom Cruise Control API authentication credentials are read. ")
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    @JsonProperty(required = true)
+    public PasswordSource getValueFrom() {
+        return valueFrom;
+    }
+
+    public void setValueFrom(PasswordSource valueFrom) {
+        this.valueFrom = valueFrom;
```

Comment:
This whole definition needs to be in the `HashLoginServiceApiUsers` class. This abstract class only handles the type (and the additional fields) -> you do not know how any other implementations will look like, whether they will reference some Secret, whether there will be some roles, users, or something to be parsed. It is also missing the subtypes. It should look very similar for example to this class: https://github.com/strimzi/strimzi-kafka-operator/blob/main/api/src/main/java/io/strimzi/api/kafka/model/common/Logging.java